agent/agent.go

```diff
@@ -3918,13 +3918,32 @@ func (a *Agent) registerCache() {
 		RefreshTimeout: 10 * time.Minute,
 	})
 
-	a.cache.RegisterType(cachetype.StreamingHealthServicesName, &cachetype.StreamingHealthServices{
-		Client: a.streamClient,
-	}, &cache.RegisterOptions{
-		Refresh:        true,
-		RefreshTimer:   0 * time.Second,
-		RefreshTimeout: 10 * time.Minute,
-	})
+	if a.config.EnableBackendStreaming {
+		a.cache.RegisterType(cachetype.StreamingHealthServicesName, &cachetype.StreamingHealthServices{
+			Client: a.streamClient,
+		}, &cache.RegisterOptions{
+			Refresh:        true,
+			RefreshTimer:   0 * time.Second,
+			RefreshTimeout: 10 * time.Minute,
+		})
+		a.cache.RegisterType(cachetype.StreamingHealthServicesConnectName, &cachetype.StreamingHealthServices{
+			Client:  a.streamClient,
+			Connect: true,
+		}, &cache.RegisterOptions{
+			Refresh:        true,
+			RefreshTimer:   0 * time.Second,
+			RefreshTimeout: 10 * time.Minute,
+		})
```

Comment:
Why have different cache types for these?

If there is only one cache type and we just use the `structs.ServiceSpecificRequest.Connect` field to switch topic inside the cache type it means less code here and at the call site where we have to make a cached request.

Reply:
I didn't realize the Connect field was part of the cache key, that's definitely cleaner.